src/charm.py

```diff
@@ -236,6 +264,65 @@ def _on_synapse_pebble_ready(self, _: ops.HookEvent, charm_state: CharmState) ->
         Args:
             charm_state: The charm state.
         """
+        logger.debug("Found %d peer unit(s).", self.peer_units_total())
+        if charm_state.redis_config is None:
+            if self.peer_units_total() > 1:
```

Comment:
I choose to add it here to make it clear that is being prevented before the configuration.

This is the order:

```
2024-03-20T15:14:55.261Z [container-agent] 2024-03-20 15:14:55 DEBUG juju-log Emitting Juju event install.
2024-03-20T15:14:55.957Z [container-agent] 2024-03-20 15:14:55 DEBUG juju-log synapse-peers:10: Emitting Juju event synapse_peers_relation_created.
2024-03-20T15:14:56.619Z [container-agent] 2024-03-20 15:14:56 DEBUG juju-log Emitting Juju event leader_settings_changed.
2024-03-20T15:14:57.265Z [container-agent] 2024-03-20 15:14:57 DEBUG juju-log Emitting Juju event synapse_pebble_ready.
2024-03-20T15:14:57.925Z [container-agent] 2024-03-20 15:14:57 DEBUG juju-log Emitting Juju event synapse_nginx_pebble_ready.
2024-03-20T15:14:59.604Z [container-agent] 2024-03-20 15:14:59 DEBUG juju-log Emitting Juju event data_storage_attached.
2024-03-20T15:15:00.248Z [container-agent] 2024-03-20 15:15:00 DEBUG juju-log Emitting Juju event config_changed.
2024-03-20T15:15:26.190Z [container-agent] 2024-03-20 15:15:26 DEBUG juju-log Emitting Juju event start.
2024-03-20T15:15:26.690Z [container-agent] 2024-03-20 15:15:26 DEBUG juju-log synapse-peers:10: Emitting Juju event synapse_peers_relation_changed.
2024-03-20T15:15:30.566Z [container-agent] 2024-03-20 15:15:30 DEBUG juju-log synapse-peers:10: Emitting Juju event synapse_peers_relation_joined.
2024-03-20T15:15:31.201Z [container-agent] 2024-03-20 15:15:31 DEBUG juju-log synapse-peers:10: Emitting Juju event synapse_peers_relation_changed.
2024-03-20T15:16:09.169Z [container-agent] 2024-03-20 15:16:09 DEBUG juju-log Emitting Juju event synapse_pebble_ready.

Reply:
What I mean is that after that, you can change a config, and another `config_changed` will be emitted, and then the charm will go to active (and the method config_changed will be run).

Reply:
Ahhhh I see.
I didn't mention it in the PR but I plan to even remove this check since when the configuration is in place, the workload will fail if there is no Redis so it could be handled that way.

For now, I could add the check in the config changed too, just in case.